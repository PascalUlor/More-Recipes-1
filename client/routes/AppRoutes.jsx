/* eslint-disable */
import React from 'react';
import {
  BrowserRouter as Router, Route, browserHistory, Switch
} from 'react-router-dom';
import HomePage from '../components/HomePage.jsx';
import SignupPage from '../components/SignupPage.jsx';
import SigninPage from '../components/SigninPage.jsx';
import Dashboard from '../components/Dashboard.jsx';
import UserRecipesPage from '../components/UserRecipesPage.jsx';
import FavoriteRecipesPage from '../components/FavoriteRecipesPage.jsx';
import RecipeDetailsPage from '../components/RecipeDetailsPage.jsx';
import Profile from '../components/Profile.jsx';
import AllRecipesPage from '../components/AllRecipesPage.jsx';
import NotFoundPage from '../components/NotFoundPage.jsx';
import RequireAuthentication from '../utils/RequireAuthentication.jsx';


const AppRoutes = () => (
  <Router history={browserHistory}>
    <Switch>
      <Route exact path="/" component={HomePage} />
      <Route exact path="/signup" component={SignupPage} />
      <Route exact path="/signin" component={SigninPage} />
      <Route exact path="/recipes" component={AllRecipesPage} />
      <Route exact path="/dashboard"
        component={RequireAuthentication(Dashboard)} />
      <Route exact path="/user/recipes"
        component={RequireAuthentication(UserRecipesPage)} />
      <Route exact path="/user/favorites"
        component={RequireAuthentication(FavoriteRecipesPage)} />
      <Route exact path="/recipes/:id/recipe-details"
        component={RecipeDetailsPage} />
      <Route exact path="/user/profile"
        component={RequireAuthentication(Profile)} />
      <Route component={NotFoundPage} />
    </Switch>
  </Router>
);

export default AppRoutes;
