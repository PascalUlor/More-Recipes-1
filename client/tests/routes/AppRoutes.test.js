/* eslint-disable */
import React from 'react';
import toJson from 'enzyme-to-json';
import AppRoutes from '../../routes/AppRoutes.jsx';


describe('<AppRoutes/>', () => {
  it('renders entire app route components without crashing', () => {
    const shallowWrapper = shallow(<AppRoutes/>);
    expect(toJson(shallowWrapper)).toMatchSnapshot();
  });
});
