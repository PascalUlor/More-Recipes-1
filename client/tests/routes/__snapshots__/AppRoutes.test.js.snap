// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppRoutes/> renders entire app route components without crashing 1`] = `
<BrowserRouter>
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/signup"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/signin"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/recipes"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/dashboard"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/user/recipes"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/user/favorites"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/recipes/:id/recipe-details"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/user/profile"
    />
    <Route
      component={[Function]}
    />
  </Switch>
</BrowserRouter>
`;
